src/spider/worker/worker.cpp

```diff
@@ -185,7 +185,48 @@ auto get_args_buffers(spider::core::Task const& task)
         );
         return std::nullopt;
     }
-    return args_buffers;
+    return arg_buffers;
+}
+
+auto get_task_details(
+        std::shared_ptr<spider::core::StorageConnection> const& conn,
+        std::shared_ptr<spider::core::MetadataStorage> const& metadata_store,
+        spider::core::TaskInstance const& instance,
+        spider::core::Task& task
+) -> bool {
+    spider::core::StorageErr const err = metadata_store->get_task(*conn, instance.task_id, &task);
+    if (!err.success()) {
+        spdlog::error("Failed to fetch task detail: {}", err.description);
+        return false;
+    }
+    return true;
+}
+
+auto setup_task(
+        std::shared_ptr<spider::core::StorageFactory> const& storage_factory,
+        std::shared_ptr<spider::core::MetadataStorage> const& metadata_store,
+        spider::core::TaskInstance const& instance,
+        spider::core::Task& task
+) -> std::optional<std::vector<msgpack::sbuffer>> {
+    std::variant<std::unique_ptr<spider::core::StorageConnection>, spider::core::StorageErr>
+            conn_result = storage_factory->provide_storage_connection();
+    if (std::holds_alternative<spider::core::StorageErr>(conn_result)) {
+        spdlog::error(
+                "Failed to connect to storage: {}",
+                std::get<spider::core::StorageErr>(conn_result).description
+        );
+        return std::nullopt;
+    }
+    std::shared_ptr const conn
+            = std::move(std::get<std::unique_ptr<spider::core::StorageConnection>>(conn_result));
+
+    // Get task details
+    if (!get_task_details(conn, metadata_store, instance, task)) {
+        return std::nullopt;
+    }
+
+    // Get task arguments
+    return get_arg_buffers(task);
```

Comment:
Adding a comment for a single function call either means the function name isn't clear or the comment is redundant.

This makes me think that `get_arg_buffers` should actually be a (public) method in `spider::core::Task`. What do you think?